go/private/actions/link.bzl

```diff
@@ -24,6 +24,30 @@ load(
     "LINKMODE_PLUGIN",
 )
 
+def _format_archive(d):
+    return "{}={}={}".format(d.label, d.importmap, d.file.path)
+
+def _map_archive(x):
+    # Build the set of transitive dependencies. Currently, we tolerate multiple
+    # archives with the same importmap (though this will be an error in the
+    # future), but there is a special case which is difficult to avoid:
+    # If a go_test has internal and external archives, and the external test
+    # transitively depends on the library under test, we need to exclude the
+    # library under test and use the internal test archive instead.
+    deps = depset(transitive = [d.transitive for d in x.archive.direct])
+    test_importmaps = {
+        t.importmap: None
+        for t in x.test_archives
+    }
+    return [
+        _format_archive(d)
+        for d in deps.to_list()
+        if d.importmap not in test_importmaps
+    ]
+
+def _map_test_archive(d):
```

Comment:
Any reason to have this as a separate function?

Reply:
No, I just thought it is more readable that way. Happy to just use `_format_archive` directly.

Reply:
I have a mild preference for using `_format_archive` directly.

I was planning to migrate these arguments into files that could be passed to `-importcfg` directly. Given what you've said about Args though, maybe it makes sense keep this and do the work in the builder, since it might not need to be done.